src/services/file/interfaces/configuration.ts

```diff
@@ -0,0 +1,15 @@
+import S3 from 'aws-sdk/clients/s3';
+
+export interface S3FileConfiguration {
```

Comment:
It's a bit a mess with the files, but some plugins uses the decorations, some get the config from the options. So to uniformize the declarations I've added it to sdk. Probably this will be encapsulated with another file namespace and won't be needed anymore in your refactor.